### DISCUSSION TITLE
lazy Objects

### DISCUSSION THREAD

**Discussion by busynest** [lit/lit#2667] [General]
@justinfagnani 
Hi Justin.
In the recent conversation it was implied that polyfills were not needed?
Compared to your recent gist: https://lit.dev/playground/#gist=b7e3c1e58657a3b130521f17b1a15af2
- LazyLoading is the only component not working for me.
- I always need to force polyfills in the index.html, and have never seen anyone talk about this:
```
if (window.customElements) window.customElements.forcePolyfill = true;
   ShadyDOM = { force: true };
   ShadyCSS = { shimcssproperties: true};
```
How could I follow your example with LazyLoader working?
in conclusion, I need all three for lit to work:
- `<script src="node_modules/@webcomponents/webcomponentsjs/webcomponents-loader.js">`
- `<script src="node_modules/lit/polyfill-support.js">`
- Forced Global Index

If I skip shadyCSS, would I have 2 apendChild?
A lazy class appendChild + A page class appendChild
```
const tag = document.createElement('lazy-loader');
tag.name = 'dynamically created';
document.body.appendChild(tag);
```

**Comment by busynest** [comment]
I mean the Lazy Class for "Page Components".

This Lazy Class:

```
import { LitElement } from 'lit';
import { property }   from 'lit/decorators.js';

export class LazyLoader extends LitElement {

  @property({type: Boolean}) public active : boolean = false;

 // Only render this page, if it is actually visible.
  shouldUpdate() { return this.active;} 
 
}

declare global {
  interface HTMLElementTagNameMap {
    'lazy-loader': LazyLoader;
  }
}
```

**Comment by busynest** [comment]
@justinfagnani 

What does shadowCSS do, that allows for Lazy Class to work on Page Components? Because display: none is important to having performance for loading, but without executing.

could we get something more official for Lazy Class that doesn’t depend on shadyCSS?

**Comment by busynest** [comment]
Therefore, I assume that I am looking for Rust-like “Lazy Behaviour / Lazy Object” for component type: “page”. As a feature of lit-applications.

Example (component types / class types) for application level:
shell
page
component
extension

the lazy class seems like an important cornerstone to lit, so looking for rust-like implementation.
